lib/chef/resource/windows_feature_powershell.rb

```diff
@@ -141,8 +141,12 @@ def to_formatted_array(x)
       action_class do
         # @return [Array] features the user has requested to install which need installation
         def features_to_install
-          # the intersection of the features to install & disabled features are what needs installing
-          @install ||= new_resource.feature_name & node["powershell_features_cache"]["disabled"]
+          # the intersection of the features to install & disabled/removed features are what needs installing
+          if new_resource.source && node["powershell_features_cache"]["removed"]
+            @install ||= new_resource.feature_name & ( node["powershell_features_cache"]["disabled"] | node["powershell_features_cache"]["removed"] )
+          else
+            @install ||= new_resource.feature_name & node["powershell_features_cache"]["disabled"]
+          end
```

Comment:
* From reading `reload_cached_powershell_data`, both of these values will always be present and will always be an array, so doing a union is safe in bot cases.
* This code looks ok, but in general memoizing a value inside a conditional can introduce subtle caching bugs. I'd invert that to make it more clear what's happening here.

```suggestion
          # the intersection of the features to install & disabled/removed features are what needs installing
          @features_to_install || = begin
            features = node["powershell_features_cache"]["disabled"]
            features |= node["powershell_features_cache"]["removed"] if new_resource.source
         
            new_resource.feature_name & features
          end
```

Reply:
Some unit tests for the different cases would be a good addition.

Reply:
Thanks @phiggins updated per suggestion!